### ISSUE TITLE
WeightedCluster.TotalWeight looks redundant 

### ISSUE THREAD

**Issue by hzxuzhonghu** [envoyproxy/envoy#21845]
```
	// Specifies the total weight across all clusters. The sum of all cluster weights must equal this
	// value, which must be greater than 0. Defaults to 100.
```

Actually this value can be inferred from all the clusters. Enforcing the management server to set this value is a bad UX. No sure why it was required.

**Comment by ggreenway** [comment]
I'm not sure what the history of this is. My first guess was that it made some algorithm less expensive, but when loading the config, there's a loop that adds up all the weights and then: 

```
    if (total_weight != total_cluster_weight_) {
      throw EnvoyException(fmt::format("Sum of weights in the weighted_cluster should add up to {}",
                                       total_cluster_weight_));
    }
```

Maybe we should make the field optional, and if not specified then don't do this check and use the sums.

@mattklein123 do you know the history of this?

**Comment by mattklein123** [comment]
> @mattklein123 do you know the history of this?

Sorry I don't remember. This might even be pre-OSS. My guess is it's to avoid a mistake in which people change the weights and don't realize what they are actually doing without knowing the entire sum. So for example, raising one and lowering another. This mistake probably happened and led to this check. :)

**Comment by mattklein123** [comment]
> That makes sense. Any objection to making it optional?

If there is a request to do so seems fine, though we might want to have a warning if it's not set? I'm not sure.

**Comment by hzxuzhonghu** [comment]
Loose a validation seems no harm to existing users.

**Comment by howardjohn** [comment]
One thing to note as well it seems TcpProxy_WeightedCluster_ClusterWeight has no TotalWeight

**Comment by ggreenway** [comment]
Fixed by #21892
